contracts/Strategy.sol

```diff
@@ -348,7 +353,9 @@ contract Strategy is BaseStrategy {
                 for (
                     uint256 i = 0;
                     i < kashiPairs.length &&
-                        sharesToFreeFromKashi > sharesFreedFromKashi;
+                        sharesToFreeFromKashi > sharesFreedFromKashi &&
+                        sharesToFreeFromKashi.sub(sharesFreedFromKashi) >
```

Comment:
question: 

does it make sense to use the threshold when withdrawing? 

this may lead to the strategy leaving dust in the strategy / returning less than requested. this should not be a real problem in this case but in a general situation I don't use the threshold for withdrawals. 

only use of threshold for withdrawals I do is to withdraw the full balance if I am leaving dust invested. 

what do you think?

Reply:
or is there any situation in which there are rounding errors and the dust will never be able to be withdrawed?

Reply:
The issue i was noticing was that rounding errors would lead us to visit additional pairs and consume extra gas when the value of visiting those pairs was low. I wanted to add an escape hatch to prevent that. I need to think about it more ...